src/main/java/org/cyclops/everlastingabilities/item/ItemAbilityTotem.java

```diff
@@ -62,6 +63,12 @@ public ItemStack getTotem(Ability ability) {
         return itemStack;
     }
 
+	public static ItemStack getRandomTotem(java.util.Random rand) {
```

Comment:
Move this to `AbilityHelpers`.

Also, instead of choosing a random `EnumRarity`, perhaps it would make more sense that the rarities of the inserted totems are taken into account for choosing the new rarity. By for example taking the max/min rarity, with a small chance of increasing the rarity. This would make this feature a bit more balanced.